plugins/outputs/iotdb/iotdb_test.go

```diff
@@ -0,0 +1,377 @@
+	testPort     = "6667"      // The server's port that you want to connect to.
+	testUser     = "root"
+	testPassword = "root"
+)
+
+func newTestClient() *IoTDB {
+	testClient := newIoTDB()
+	testClient.Host = testHost
+	testClient.Port = testPort
+	testClient.User = testUser
+	testClient.Password = testPassword
+	testClient.Log = testutil.Logger{}
+	return testClient
+}
+
+func TestConnectAndClose(t *testing.T) {
+	if testing.Short() {
+		t.Skip("Skipping integration test in short mode")
+	}
+	testClient := newTestClient()
+
+	var err error
+	err = testClient.Connect()
+	require.NoError(t, err)
+	err = testClient.Close()
+	require.NoError(t, err)
+}
+
+func TestInitAndConnect(t *testing.T) {
+	if testing.Short() {
+		t.Skip("Skipping integration test in short mode")
+	}
+	testClient := newTestClient()
+
+	var err error
+	err = testClient.Init()
+	require.NoError(t, err)
+	err = testClient.Connect()
+	require.NoError(t, err)
+	err = testClient.Close()
+	require.NoError(t, err)
+}
+
+func generateTestMetric(
+	name string,
+	tags []telegraf.Tag,
+	fields []telegraf.Field,
+	timestamp time.Time,
+) telegraf.Metric {
+	m := metric.New(name, map[string]string{}, map[string]interface{}{}, timestamp)
+	for _, tag := range tags {
+		m.AddTag(tag.Key, tag.Value)
+	}
+	for _, field := range fields {
+		m.AddField(field.Key, field.Value)
+	}
+	return m
+}
+
+var (
+	constTestTimestamp = time.Date(2022, time.July, 20, 12, 25, 33, 44, time.UTC)
+	testMetrics001     = []telegraf.Metric{
+		generateTestMetric(
+			"root.computer.fan",
+			[]telegraf.Tag{
+				{Key: "price", Value: "expensive"},
+				{Key: "owner", Value: "cpu"},
+			},
+			[]telegraf.Field{
+				{Key: "temperature", Value: float64(42.55)},
+				{Key: "counter", Value: int64(987654321)},
+			},
+			constTestTimestamp,
+		),
+		generateTestMetric(
+			"root.computer.fan",
+			[]telegraf.Tag{ // same keys in different order
+				{Key: "owner", Value: "gpu"},
+				{Key: "price", Value: "cheap"},
+			},
+			[]telegraf.Field{
+				{Key: "temperature", Value: float64(56.24)},
+				{Key: "counter", Value: int64(123456789)},
+			},
+			constTestTimestamp,
+		),
+		generateTestMetric(
+			"root.computer.keyboard",
+			[]telegraf.Tag{},
+			[]telegraf.Field{
+				{Key: "temperature", Value: float64(30.33)},
+				{Key: "counter", Value: int64(123456789)},
+				{Key: "unsigned_big", Value: uint64(math.MaxInt64 + 1000)},
+				{Key: "string", Value: "Made in China."},
+				{Key: "bool", Value: bool(false)},
+				{Key: "int_text", Value: "123456789011"},
+			},
+			constTestTimestamp,
+		),
+	}
+	testMetrics002 = []telegraf.Metric{
+		generateTestMetric(
+			"root.computer.mouse",
+			[]telegraf.Tag{},
+			[]telegraf.Field{
+				{Key: "unsigned_big", Value: uint64(math.MaxInt64 + 1000)},
+			},
+			constTestTimestamp,
+		),
+	}
+)
+
+// compare two recordsWithTags, returns True if and only if they are the same.
+func compareRecords(rwt1 *recordsWithTags, rwt2 *recordsWithTags, log telegraf.Logger) bool {
```

Comment:
Can't you use [`require.EqualValues`](https://pkg.go.dev/github.com/stretchr/testify/require#EqualValues) instead of hand-coding this?

Reply:
I use `compareRecords` because TagList is not important, so I don't need to compare their whole content.
But you remind me to use a better way to compare them, I will use `require.EqualValues` separately for each list.